llvm/utils/TableGen/X86DisassemblerTables.cpp

```diff
@@ -785,9 +785,9 @@ void DisassemblerTables::emitModRMDecision(raw_ostream &o1, raw_ostream &o2,
     break;
   }
 
-  // We assume that the index can fit into uint16_t.
-  assert(sEntryNumber < 65536U &&
-         "Index into ModRMDecision is too large for uint16_t!");
+  // We assume that the index can fit into uint32_t.
+  assert(sEntryNumber < -1U &&
+         "Index into ModRMDecision is too large for uint32_t!");
```

Comment:
I suggest not using the tricky `-1`, it's not robust b/c the suffix `U` can represent unsigned long long (https://en.cppreference.com/w/cpp/language/integer_literal)

Reply:
`-1U` is wildly used in LLVM code base. I think the document means it can be automatically recognized as `LLU` with the range of constant value, instead of not to use it for small value. See https://godbolt.org/z/9W9srbYYM